packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -178,6 +178,18 @@ class RequestWrapper {
   async fetch({request}) {
     assert.atLeastOne({request});
 
+    if (this.behaviorCallbacks.requestWillFetch) {
+      for (let callback of this.behaviorCallbacks.requestWillFetch) {
+        const returnedRequest = callback({request});
+        if (returnedRequest && returnedRequest.then) {
```

Comment:
makes sense 👍 
changed code to use `assert` and now only expect the callback to return a promise